pkgs/applications/blockchains/dogecoin/default.nix

```diff
@@ -0,0 +1,104 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pkg-config,
+  autoreconfHook,
+  db5,
+  openssl,
+  boost,
+  zlib,
+  miniupnpc,
+  libevent,
+  protobuf,
+  qtbase ? null,
+  wrapQtAppsHook ? null,
+  qttools ? null,
+  qmake ? null,
+  qrencode,
+  withGui,
+  withUpnp ? false,
+  withUtils ? true,
+  withWallet ? true,
+  withZmq ? true,
+  zeromq,
+  util-linux ? null,
+  Cocoa ? null,
+}:
+
+stdenv.mkDerivation rec {
```

Comment:
I think we should use the `finalAttrs` pattern here.

The `finalAttrs` pattern will let you remove the `rec` keyword ([see its implementation in Nix](https://github.com/NixOS/nix/blob/2d1d81114d72ace89ce08cd3bc93f4eb27a2975d/src/libexpr/nixexpr.cc#L121)).

Why is this pattern preferred to `rec` ?

Let's take this simple code example:

```nix
mkDerivation rec {
   foo = 1;
   bar = foo + 1;
}
```

and then `.overrideAttrs(old: { foo = 2; })`, you'll get `{ foo = 2; bar = 2; }` while with `finalAttrs` pattern, it would work correctly because it's a real fixed point.

Let me share a couple of useful links regarding the `finalAttrs` pattern:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Feel free to reach out if you need some assistance.

Reply:
i didnt really write that rec part, and i do personally try to avoid it

if you look at each commit seperately, youll see the first one is a revert of somebody deleting dogecoin entirely, which is where the rec came from
and the next commit is just updating the version, and disabling upnp

Reply:
> i didnt really write that rec part, and i do personally try to avoid it
> if you look at each commit seperately, youll see the first one is a revert of somebody deleting dogecoin entirel

You could argue that `rec` is not worse than `finalAttrs` (which [I might say](https://github.com/NixOS/nixpkgs/issues/310373)). But your reason does not seem valid to me. "It is written by someone else" or "it used to be in nixpkgs" does not imply it meets the current practice. Legacy means legacy for a reason.

This should be reviewed as a new-package PR, because it adds a package which did not exist. You could reference the previous commit, but now that code is under your contribution and you are responsible for it.